### DISCUSSION TITLE
Why are 64bit integers json serialised as strings?

### DISCUSSION THREAD

**Discussion by mira01** [tomas-abrahamsson/gpb#223] [Q&A] [answered]
Hello,
What is the reason for different json serialisation of 32 bit and 64 bit integers? Why are 64bit integers serialised to json with quotes as strings?
[https://github.com/tomas-abrahamsson/gpb/blob/master/src/gpb_gen_json_encoders.erl#L612](https://github.com/tomas-abrahamsson/gpb/blob/master/src/gpb_gen_json_encoders.erl#L612)
Thank you

**Comment by tomas-abrahamsson** [comment, marked as answer]
The [Json mapping section in the langage guide](https://developers.google.com/protocol-buffers/docs/proto3#json) says that the int64 types are to be represented in JSON as a string: "JSON value will be a decimal string." I suppose this is to not lose accuracy when large values are represented as floats.
